java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -103,73 +121,83 @@ public final String getFullName() {
     /**
      * Gets the {@link Class} instance for the Java primitive values (non-objects) of this type
      * or null if a Java primitive type does not exist. This method complements the
-     * {@link BondType#getValueClass()} method which returns the classes for object instances.
+     * {@link #getValueClass()} method which returns the classes for object instances.
+     * Thus, for example, this method returns {@link Long#TYPE} for Bond int64 data type whereas
+     * the other method returns the class object for the {@link Long} boxed type.
      *
      * @return the class object for a primitive type or null if a primitive type doesn't exists
      */
     public abstract Class<T> getPrimitiveValueClass();
 
     /**
      * Gets a value indicating whether values of this type can be assigned to null.
+     * In Bond, only values explicitly declared as nullable can be assigned to the null value.
      *
      * @return whether value of this type can be assigned to null
      */
     public abstract boolean isNullableType();
 
     /**
-     * Indicates whether this type is a generic type that is specialized with one or more generic type arguments.
+     * Indicates whether this type is a generic type (either a native Bond container or a user-defined struct)
+     * that is specialized with one or more generic type arguments. This method returns true if and only if
+     * the {@link #isGenericType()} method returns a non-empty array with one or more elements.
      *
      * @return true if this type is a specialization of a generic type
      */
     public abstract boolean isGenericType();
 
     /**
-     * Gets a new array instance containing the type descriptors of the generic type arguments or null
-     * if this type is not a specialization of a generic type.
+     * Gets a new array instance containing the type descriptors of one or more generic type arguments or null
+     * if this type is not a specialization of a generic type. This method returns null if the type is not
+     * generic (i.e. if the {@link #isGenericType() returns false). Please note that this method never returns
+     * an empty array since a generic type must have at least one unbound generic type parameter.
      *
-     * @return an array containing the descriptors of the generic type arguments or null
+     * @return an array containing the descriptors of the generic type arguments or null if not a generic type
      */
     public abstract BondType<?>[] getGenericTypeArguments();
 
     /**
      * Returns the default value of this type as a shared instance for immutable primitive types
      * or a new instance for all other types. For non-nullable structs and containers this method
-     * returns an initialized value equivalent to invoking the default constructor. For nullable
-     * values this method returns null.
+     * returns an initialized value equivalent to invoking the public constructor. For nullable
```

Comment:
Can you clarify what nullary means? I updated the comment to specify which constructor (it's not always the parameterless one).

---
In reply to: [124082288](https://github.com/Microsoft/bond/pull/502#discussion_r124082288) [](ancestors = 124082288)